connectivity/suite.go

```diff
@@ -240,6 +275,124 @@ func Run(ctx context.Context, ct *check.ConnectivityTest) error {
+				// Trying to access private endpoint without "secret" header set
+				// should lead to a drop.
+				if a.Destination().Path() == "/private" && !a.Destination().HasLabel("X-Very-Secret-Token", "42") {
+					return check.ResultDropCurlHTTPError, check.ResultNone
+				}
+				egress = check.ResultOK
+				// Expect all curls from client2 to be proxied and to be GET calls.
+				egress.HTTP = check.HTTP{
+					Method: "GET",
+				}
+				return egress, check.ResultNone
+			}
+			return check.ResultDrop, check.ResultNone
+		})
+
+	// Tests with deny policy
+	if ct.DenyPolicy() {
+		// This policy denies ingress into client from client2.
+		ct.NewTest("client-ingress-deny").WithPolicy(clientIngressFromClient2DenyPolicyYAML).
+			WithScenarios(
+				tests.ClientToClient(),
+			).
+			WithExpectations(func(a *check.Action) (egress, ingress check.Result) {
+				if a.Destination().HasLabel("kind", "client") {
+					return check.ResultOK, check.ResultDrop
+				}
+				return check.ResultOK, check.ResultOK
+			})
+
+		// This policy allows ingress to echo only from client with a label 'other:client'
+		ct.NewTest("echo-ingress-deny").WithPolicy(echoIngressFromOtherClientDenyPolicyYAML).
+			WithScenarios(
+				tests.PodToPod(),
+			).
+			WithExpectations(func(a *check.Action) (egress, ingress check.Result) {
+				if a.Destination().HasLabel("kind", "echo") {
+					return check.ResultDrop, check.ResultDrop
+				}
+				return check.ResultOK, check.ResultOK
+			})
+
+		// This policy denies ingress to echo only from client with a label 'other:client'.
+		ct.NewTest("client-ingress-icmp-deny").WithPolicy(echoIngressICMPDenyPolicyYAML).
+			WithFeatureRequirements(check.RequireFeatureEnabled(check.FeatureICMPPolicy)).
+			WithScenarios(
+				tests.ClientToClient(),
+			).
+			WithExpectations(func(a *check.Action) (egress, ingress check.Result) {
+				if a.Destination().HasLabel("kind", "client") {
+					return check.ResultOK, check.ResultDrop
+				}
+				return check.ResultOK, check.ResultOK
+			})
+
+		// This policy denies port 8080 from client to echo
+		ct.NewTest("client-egress-deny").WithPolicy(clientEgressToEchoDenyPolicyYAML).
+			WithScenarios(
+				tests.PodToPod(),
+			).
+			WithExpectations(func(a *check.Action) (egress, ingress check.Result) {
+				if a.Destination().HasLabel("kind", "echo") &&
+					a.Source().HasLabel("kind", "client") {
+					return check.ResultDrop, check.ResultNone
+				}
+				return check.ResultOK, check.ResultOK
+			})
+
+		// This policy denies port 8080 from client (using expression) to echo
+		ct.NewTest("client-egress-expression-deny").WithPolicy(clientEgressToEchoExpressionDenyPolicyYAML).
+			WithScenarios(
+				tests.PodToPod(),
+			).
+			WithExpectations(func(a *check.Action) (egress, ingress check.Result) {
+				if a.Destination().HasLabel("kind", "echo") &&
+					a.Source().HasLabel("kind", "client") {
+					return check.ResultDrop, check.ResultNone
+				}
+				return check.ResultOK, check.ResultOK
```

Comment:
I got a bit confused by this default allow, but I think pod-to-pod only ever sends traffic from client -> echo so all cases are covered in the condition above, right? So I guess this code is never hit.

Reply:
yeah, I guess partially it was me, who didn't know about the default deny :man_facepalming: 

I will update the default behaviour to Drop for both Ingress and Egress.